File: 151051/Varanko/app/src/main/java/by/bsuir/dc/impl/author/model/Author.java
package by.bsuir.dc.impl.author.model;

import by.bsuir.dc.api.base.AbstractEntity;
import by.bsuir.dc.impl.story.model.Story;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "tbl_author", schema = "distcomp")
@RequiredArgsConstructor
@NoArgsConstructor
public class Author extends AbstractEntity {
    @Column(unique=true)
    @NonNull String login;
    @NonNull String password;
    @NonNull String firstname;
    @NonNull String lastname;

    @OneToMany(mappedBy = "author", orphanRemoval = true)
    private List<Story> stories = new ArrayList<>();
}

File: 251004/Ubozhenko/DistComp_lab_1/DistComp_1/Models/Author.cs
namespace DistComp_1.Models;

public class Author : BaseModel
{
    public string Login { get; set; }
    public string Password { get; set; }
    public string Firstname { get; set; }
    public string Lastname { get; set; }

    public List<News> Stories { get; set; } = [];
}